File: shared-packages/packages/rubot-logger/rubot_logger/envelope.py
from __future__ import annotations

from datetime import datetime, timezone
from typing import Any, Literal

from pydantic import BaseModel, Field

LogLevel = Literal["DEBUG", "INFO", "WARN", "ERROR"]


class ErrorDetail(BaseModel):
    type: str
    message: str
    stack: str | None = None


class LogEnvelope(BaseModel):
    timestamp: str = Field(
        default_factory=lambda: datetime.now(timezone.utc).isoformat(),
    )
    log_level: LogLevel
    service: str
    component: str
    environment: str
    deployment_hash: str = ""
    tenant_id: str | None = None
    sender_id: str | None = None
    chat_source_session_id: str | None = None
    trace_id: str | None = None
    span_id: str | None = None
    event_type: str
    message: str
    error: ErrorDetail | None = None
    extra: dict[str, Any] = Field(default_factory=dict)
    agent: dict[str, Any] | None = None

File: shared-packages/packages/rubot-logger-ts/src/envelope.ts
export type LogLevel = "DEBUG" | "INFO" | "WARN" | "ERROR";

export interface ErrorDetail {
  type: string;
  message: string;
  stack?: string;
}

export interface LogEnvelope {
  timestamp: string;
  log_level: LogLevel;
  service: string;
  component: string;
  environment: string;
  deployment_hash: string;
  tenant_id?: string | null;
  sender_id?: string | null;
  chat_source_session_id?: string | null;
  trace_id?: string | null;
  span_id?: string | null;
  event_type: string;
  message: string;
  error?: ErrorDetail | null;
  extra?: Record<string, unknown>;
  agent?: Record<string, unknown> | null;
}
